packages/@webex/plugin-meetings/src/roap/turnDiscovery.ts

```diff
@@ -284,30 +340,46 @@ export default class TurnDiscovery {
       };
     }
 
-    return this.sendRoapTurnDiscoveryRequest(meeting, isReconnecting)
-      .then(() => this.waitForTurnDiscoveryResponse())
-      .then(() => this.sendRoapOK(meeting))
-      .then(() => {
-        this.defer = undefined;
+    try {
+      const httpResponse = await this.sendRoapTurnDiscoveryRequest(meeting, isReconnecting);
 
-        LoggerProxy.logger.info('Roap:turnDiscovery#doTurnDiscovery --> TURN discovery completed');
+      // if we haven't got the response over http, we need to wait for it to come over the websocket via Mercury
+      const {isOkRequired} = httpResponse
+        ? await this.handleTurnDiscoveryResponseInHttpResponse(httpResponse)
+        : await this.waitForTurnDiscoveryResponse();
+
+      if (isOkRequired) {
+        await this.sendRoapOK(meeting);
 
-        return {turnServerInfo: this.turnInfo, turnDiscoverySkippedReason: undefined};
-      })
-      .catch((e) => {
-        // we catch any errors and resolve with no turn information so that the normal call join flow can continue without TURN
         LoggerProxy.logger.info(
-          `Roap:turnDiscovery#doTurnDiscovery --> TURN discovery failed, continuing without TURN: ${e}`
+          'Roap:turnDiscovery#doTurnDiscovery --> TURN discovery response requires OK'
         );
 
-        Metrics.sendBehavioralMetric(BEHAVIORAL_METRICS.TURN_DISCOVERY_FAILURE, {
+        Metrics.sendBehavioralMetric(BEHAVIORAL_METRICS.TURN_DISCOVERY_REQUIRES_OK, {
```

Comment:
question: I know our handling of this response header was only recently added, do we expect to not receive it sometimes? or is this code here "just in case"

Reply:
Yeah the support in Linus for this was GA-ed just a week or 2 ago, so in theory we should always receive "noOkInTransaction", but I want to confirm that with our metrics and then we can later on remove the code for sending OK completely. The code for sending OK is kept also in case there is some problem and Linus rolls back their toggle.